Deploying Machine Learning models using Flask

srishilesh opened the pull request:
### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #869

lalith1403 commented on articles/deploying-machine-learning-models-using-flask/index.md:
```diff
@@ -0,0 +1,299 @@
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/deploying-machine-learing-models-using-flask/hero.jpg
```
check all the URLs carefully. It says learing, but I think it should be learning, looking at the folder name.

srishilesh replied:
Thanks for that.

lalith1403 commented on articles/deploying-machine-learning-models-using-flask/index.md:
````diff
@@ -0,0 +1,299 @@
+
+```python
+# Importing the dataset
+data = pd.read_csv('[YOUR-ABSOLUTE-PATH]/iris.csv')
````
This might cause confusion to absolute beginners. Instead, use a relative path and ask them to store both the code and csv file in the same directory.

srishilesh replied:
Done

lalith1403 reviewed (changes requested):
Well written article. A few minor changes are required. I will check the code once, you make the changes.

srishilesh commented:
Hi @lalith1403 . Thanks for the review. I have made the requested changes.

lalith1403 commented:
I got this error while running the `LogisticRegression` model.

```txt
ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.
```  
I made the following changes to the `model.py` file.

```py
logreg = LogisticRegression(max_iter=1000) # Initializing the Logistic Regression model
logreg.fit(X, y) # Fitting the model
```

Then, I ran the command, 
```py
python server.py
```

The server is not getting initialized. Can you rectify the above two points in the article. Also, you should specify the name of the file you are creating and where it is saved,

lalith1403 commented:
> I got this error while running the `LogisticRegression` model.
> 
> ```
> ConvergenceWarning: lbfgs failed to converge (status=1):
> STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.
> ```
> 
> I made the following changes to the `model.py` file.
> 
> ```python
> logreg = LogisticRegression(max_iter=1000) # Initializing the Logistic Regression model
> logreg.fit(X, y) # Fitting the model
> ```
> 
> Then, I ran the command,
> 
> ```python
> python server.py
> ```
> 
> The server is not getting initialized. Can you rectify the above two points in the article? Also, you should specify the name of the file you are creating and where it is saved,

You have forgotten to run the app. Use `app.run()`

lalith1403 reviewed (changes requested):
Also, mention the directory structure of the files present. For example:
```txt
--| templates
------| home.html
------| output.html
---| iris.csv 
```

srishilesh commented:
Hi @lalith1403 . Thanks for the Tech review. I have made the following requested changes.
- Fixed the warning in `LogisticRegression` as you mentioned.
- Added `app.run()` with explanation.
- Name of the files while creation has been mentioned.
- File directory structure has been added.

Please let me know if I have to make anymore changes.

lalith1403 commented:
@srishilesh Check the code once before submitting, to avoid multiple rounds of revisions. I had found a few errors in your previous article, as well.

srishilesh commented:
> @srishilesh Check the code once before submitting, to avoid multiple rounds of revisions. I had found a few errors in your previous article, as well.

@lalith1403 Thanks for the review. I will make sure that I review my code before submission.

hectorkambow commented:
plagiarism check done
